### PR TITLE
Optimize other index

### PR DESCRIPTION
**Pull request by mnunberg**

### CODE REVIEW

**Comment by mnunberg** [comment]
Actually, there's some stuff that can be moved out of this PR

File: src/indexer.c [hunk 1]
```diff
@@ -139,7 +142,7 @@ static int writeMergedEntries(DocumentIndexer *indexer, RSAddDocumentCtx *aCtx,
 
   // This is used as a cache layer, so that we don't need to derefernce the
   // RSAddDocumentCtx each time.
-  uint32_t docIdMap[MAX_DOCID_ENTRIES] = {0};
+  uint32_t docIdMap[MAX_BULK_DOCS] = {0};
```
Comment (MeirShpilraien) [inline src/indexer.c]: maybe its better to use the arr object in util for those two

File: src/indexer.c [hunk 2]
```diff
@@ -291,6 +294,53 @@ static void doAssignIds(RSAddDocumentCtx *cur, RedisSearchCtx *ctx) {
   }
 }
 
+static void indexBulkFields(RSAddDocumentCtx *aCtx, RedisSearchCtx *sctx) {
+  // Traverse all fields, seeing if there may be something which can be written!
+  IndexBulkData bData[SPEC_MAX_FIELDS] = {{0}};
+  IndexBulkData *activeBulks[SPEC_MAX_FIELDS];
```
Comment (MeirShpilraien) [inline src/indexer.c]: maybe its better to use the arr object in util for those two

Comment (mnunberg) [reply src/indexer.c]: Why use a dynamic array when I can use a static one?

Comment (MeirShpilraien) [reply src/indexer.c]: there will be no need to save the size and increasing it, just a suggestion ...

**Pull request merged**